### ISSUE TITLE
Error when setting master DatePicker value through setValue

### ISSUE THREAD

**Issue by kyubisation** [SchweizerischeBundesbahnen/sbb-angular#15]
Copied from https://issues-ext.sbb.ch/browse/SACL-155

Error when setting master DatePicker value through FormControl: When using the DatePicker in the master/slave setup and setting the value of the master through FormControl.setValue(new Date()) the following error is logged:

```
ERROR TypeError: Cannot read property 'value' of undefined
```

**Comment by kyubisation** [comment]
Unable to reproduce

**Comment by daniel-sc** [comment]
The following snipptes should reproduce this bug:
Template:
```html
  <sbb-field mode="long">
    <sbb-label for="date_from">von</sbb-label>
    <sbb-datepicker [slave]="dateTo">
      <input type="text" sbbDateInput name="date_from" id="date_from" [formControl]="dateFrom">
    </sbb-datepicker>
  </sbb-field>
  <sbb-field mode="long">
    <sbb-label for="date_to">bis</sbb-label>
    <sbb-datepicker #date_to>
      <input type="text" sbbDateInput name="date_to" id="date_to" [formControl]="dateTo">
    </sbb-datepicker>
  </sbb-field>
```
Controller:
```typescript
@Component({
  selector: 'app-date-filter',
  templateUrl: './date-filter.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class DateFilterComponent implements OnInit, OnChanges {

  @Input()
  fromValue: Date;
  @Input()
  toValue: Date;

  @Output()
  filterChanged = new EventEmitter<{ from?: Date, to?: Date }>();

  dateFrom = new FormControl();
  dateTo = new FormControl();

  constructor() {
  }

  ngOnInit() {
    merge(this.dateFrom.valueChanges, this.dateTo.valueChanges).pipe(
      map(() => ({from: this.dateFrom.value, to: this.dateTo.value})),
      distinctUntilChanged((a, b) => CompareHelper.equalsObject(a, b)),
    ).subscribe(newFilter => this.filterChanged.emit(newFilter));
  }

  ngOnChanges(changes: SimpleChanges): void {
    if (changes.fromValue) {
      this.dateFrom.setValue(this.fromValue);
    }
    if (changes.toValue) {
      this.dateTo.setValue(this.toValue);
    }
  }

}
```

Using this component with something like
```html
<app-date-filter [fromValue]="now" [toValue]="null"></app-date-filter>
```
reproduces the bug.

**Comment by daniel-sc** [comment]
Probably there is some issue in my code.. I'll investigate further..

**Comment by daniel-sc** [comment]
As far as I see, this is a life cycle problem. The error occurs only when the value is changed "early" - e.g. in the `onInit` method. Does this help reproducing it?

**Comment by kyubisation** [comment]
Could you provide the full error log?
At first glance this does not seem to be a problem of this library, but it is not quite clear.

**Comment by daniel-sc** [comment]
![image](https://user-images.githubusercontent.com/117919/60001256-67929100-9666-11e9-8073-55e2790a7310.png)
And as well as text: 
[localhost-1561362920096.log](https://github.com/SchweizerischeBundesbahnen/sbb-angular/files/3319800/localhost-1561362920096.log)

**Comment by kyubisation** [comment]
@daniel-sc Can you provide a reproduction with StackBlitz? https://stackblitz.com/fork/angular-axrgm3
This would help me find the error faster.

**Comment by daniel-sc** [comment]
@kyubisation https://stackblitz.com/edit/angular-klskaj

**Comment by kyubisation** [comment]
@daniel-sc Thank you very much for the reproduction. This should be fixed by #45. I'll revisit this, once I publish the new release (today).

**Comment by kyubisation** [comment]
Closed by #45
